mlrun/datastore/targets.py

```diff
@@ -1467,6 +1471,256 @@ def as_df(
+        :param collection_name:             the name of the collection to access,
+                                            from the current database
+        :param schema:                      the schema of the collection (must pass when
+                                            create_collection=True)
+        :param primary_key_column:          the primary key of the collection (must pass always)
+        :param if_exists:                   {'fail', 'replace', 'append'}, default 'fail'
+                                            - fail: If table exists, do nothing.
+                                            - replace: If table exists, drop it, recreate it, and insert data.
+                                            - append: If table exists, insert data. Create if does not exist.
+        :param create_collection:           pass True if you want to create new collection named by
+                                            collection_name with schema on current database.
+        :param create_according_to_data:    (not valid)
+        """
+        create_according_to_data = False  # TODO: open for user
+        db_path = db_path or os.getenv(self._SQL_DB_PATH_STRING_ENV_VAR)
+        if db_path is None or collection_name is None:
+            attr = {}
+        else:
+            # check for collection existence and acts according to the user input
+            self._primary_key_column = primary_key_column
+            import sqlalchemy as db
+
+            engine = db.create_engine(db_path)
+            sql_connection = engine.connect()
+            metadata = db.MetaData()
+            collection_exists = engine.dialect.has_table(
+                sql_connection, collection_name
+            )
+            if not collection_exists and not create_collection:
+                raise ValueError(f"Collection named {collection_name} is not exist")
+
+            elif not collection_exists and create_collection:
+                TYPE_TO_SQL_TYPE = {
+                    int: db.Integer,
+                    str: db.String,
+                    datetime.datetime: db.DateTime,
+                    pd.Timestamp: db.DateTime,
+                    bool: db.Boolean,
+                    float: db.Float,
+                }
+                # creat new collection with the given name
+                columns = []
+                for col, col_type in schema.items():
+                    col_type = TYPE_TO_SQL_TYPE.get(col_type)
+                    if col_type is None:
+                        raise TypeError(f"{col_type} unsupported type")
+                    columns.append(
+                        db.Column(
+                            col, col_type, primary_key=(col == primary_key_column)
+                        )
+                    )
+
+                db.Table(collection_name, metadata, *columns)
+                metadata.create_all(engine)
```

Comment:
I don't think it's right to create a SQL table inside of the constructor. Table creation should happen just before writing. The way it is now, simple creating a `SqlDBTarget` will try to create the SQL table, and that shouldn't happen until the user runs an operation (like `ingest`).